### PR TITLE
Create separate provider for the context

### PR DESCRIPTION
**Pull request by TimothyAsirJeyasing**
Create separate provider for the context and pass its descendants as a children
https://github.com/red-hat-storage/odf-console/issues/672

### CODE REVIEW

**Comment by TimothyAsirJeyasing** [comment]
@SanjalKatiyar Please review

File: packages/odf/components/topology/Topology.tsx [hunk 1]
```diff
@@ -531,24 +531,64 @@ const Topology: React.FC = () => {
...
+  const TopologySearchContextProvider: React.FC<{ children: React.ReactNode }> =
+    ({ children }) => {
+      const contextValue = React.useMemo(
+        () => ({
+          activeItemsUID,
+          setActiveItemsUID,
+          activeItem,
+          setActiveItem,
+        }),
+        [activeItemsUID, setActiveItemsUID, activeItem, setActiveItem]
```
Comment (bipuladh) [inline packages/odf/components/topology/Topology.tsx:579]: Please resolve these dependency warnings.

Comment (TimothyAsirJeyasing) [reply packages/odf/components/topology/Topology.tsx:568]: Done

**Comment by bipuladh** [comment]
build tests are failing

File: packages/odf/components/topology/Topology.tsx [hunk 2]
```diff
@@ -531,24 +531,53 @@ const Topology: React.FC = () => {
 
   const zones = memoizedNodes.map(getTopologyDomain);
 
+  const TopologyDataContextProvider: React.FC<{ children: React.ReactNode }> =
```
Comment (SanjalKatiyar) [inline packages/odf/components/topology/Topology.tsx:534]: why are creating nested FCs ?

Comment (SanjalKatiyar) [reply packages/odf/components/topology/Topology.tsx:534]: create sperate provider...

File: packages/odf/components/topology/Topology.tsx [hunk 3]
```diff
@@ -531,24 +531,53 @@ const Topology: React.FC = () => {
...
+      );
+
+      return (
+        <TopologyDataContext.Provider value={contextValue}>
+          {children}
+        </TopologyDataContext.Provider>
+      );
+    };
+
+  const TopologySearchContextProvider: React.FC<{ children: React.ReactNode }> =
```
Comment (SanjalKatiyar) [inline packages/odf/components/topology/Topology.tsx:559]: create sperate provider... do not do nesting of FCs...

**Comment by SanjalKatiyar** [comment]
is this PR tested ?
please add SS for both MCO and ODF pages where we have made changes...

File: packages/odf/components/topology/Topology.tsx (hunk 2 above)
Comment (TimothyAsirJeyasing) [reply packages/odf/components/topology/Topology.tsx:534]: Could you please give some more light

File: packages/odf/components/topology/Topology.tsx (hunk 1 above)
Comment (SanjalKatiyar) [reply packages/odf/components/topology/Topology.tsx:568]: please do not resolve comments unless it is fixed... I can still see the same warnings...

File: packages/odf/components/topology/Topology.tsx (hunk 2 above)
Comment (SanjalKatiyar) [reply packages/odf/components/topology/Topology.tsx:534]: do not create a FC inside another FC...

File: packages/mco/components/mco-dashboard/data-policy/dr-dashboard-context.tsx [hunk 4]
```diff
@@ -2,6 +2,21 @@ import * as React from 'react';
...
+          <AccumulatedComponents>
+            <CurrentComponent>{children}</CurrentComponent>
+          </AccumulatedComponents>
+        );
+      };
+    },
+    ({ children }) => <>{children}</>
+  );
+};
+
```
Comment (SanjalKatiyar) [inline packages/mco/components/mco-dashboard/data-policy/dr-dashboard-context.tsx:19]: no need for this...

File: packages/mco/components/mco-dashboard/data-policy/dr-dashboard.tsx [hunk 5]
```diff
@@ -159,21 +179,28 @@ export const DRDashboard: React.FC = () => {
...
+  return (
+    <DRResourcesContext.Provider value={contextValue}>
+      {children}
+    </DRResourcesContext.Provider>
+  );
+};
+
+const providers = [CSVStatusesContextProvider, DRResourcesContextProvider];
+
+const AppContextProvider = combineComponents(...providers);
```
Comment (SanjalKatiyar) [inline packages/mco/components/mco-dashboard/data-policy/dr-dashboard.tsx:196]: no need for this right now...

File: packages/mco/components/mco-dashboard/data-policy/dr-dashboard.tsx [hunk 6]
```diff
@@ -159,21 +179,28 @@ export const DRDashboard: React.FC = () => {
...
 
-  // ToDo(Sanjal): combime multiple Context together to make it scalable
-  // refer: https://javascript.plainenglish.io/how-to-combine-context-providers-for-cleaner-react-code-9ed24f20225e
```
Comment (SanjalKatiyar) [inline packages/mco/components/mco-dashboard/data-policy/dr-dashboard.tsx:169]: add this comment back...

**Comment by TimothyAsirJeyasing** [comment]
@SanjalKatiyar Please review

**Comment by TimothyAsirJeyasing** [comment]
After encountering many changes and merge conflicts, I am closing this and I will be sending a separate pull request.

/close

**Pull request closed without merging**